scene/gui/label.cpp

```diff
@@ -748,9 +748,13 @@ void Label::set_text(const String &p_string) {
 	text = p_string;
 	xl_text = atr(p_string);
 	dirty = true;
-	if (visible_ratio < 1) {
-		visible_chars = get_total_character_count() * visible_ratio;
+
+	if (visible_chars >= 0) { // addresses issue discussed in https://github.com/godotengine/godot/issues/83387#issuecomment-1763396697		
+		// We recalculate the ratio when the string changes. It is more intuitive for users to peg the visible characters value rather than the ratio, 
```

Comment:
```suggestion
		// We recalculate the ratio when the string changes. It is more intuitive for users to peg the visible characters value rather than the ratio,
```